rules/rbe_repo.bzl

```diff
@@ -0,0 +1,597 @@
+
+  2 - When output_base is not set (env var "RBE_AUTOCONF_ROOT" is not required),
+    running this rule will create targets in the
+    remote repository (e.g., rbe_default) which can be used to point your
+    flags to:
+
+      bazel build ... \ 
+                --crosstool_top=@rbe_default//rbe_config_cc:toolchain \
+                --host_javabase=@rbe_default//platforms:jdk8 \
+                --javabase=@rbe_default//platforms:jdk8 \
+                --host_java_toolchain=@bazel_tools//tools/jdk:toolchain_hostjdk8 \
+                --java_toolchain=@bazel_tools//tools/jdk:toolchain_hostjdk8 \
+                --extra_execution_platforms=@rbe_default//platforms:rbe_ubuntu1604 \
+                --host_platform=@rbe_default//platforms:rbe_ubuntu1604 \
+                --platforms=@rbe_default//platforms:rbe_ubuntu1604 \
+                --extra_toolchains=@rbe_default//platforms:cc-toolchain \
+
+    Note running bazel clean --expunge_async, or otherwise modifying attrs or
+    env variables used by this rule will trigger it to re-execute. Running this
+    repo rule takes some time as it needs to pull a container, run it, and then
+    run some commands inside. We recommend you use output_base and check in the produced
+    files so you dont need to run this rule with every clean build.
+
+The {bazel_version} above corresponds to the version of bazel installed locally.
+Note you can override this version and pass an optional rc# if desired.
+Running this rule with a non release version (e.g., built from source) will not work.
+If running with bazel built from source you must pass a bazel_version and bazel_rc
+to rbe_autoconfig. Also, note the bazel_version bazel_rc must be published in 
+https://releases.bazel.build/...
+
+Note this is a very not hermetic repository rule that can actually change the
+contents of your project sources. While this is generally not recommended by
+Bazel, its the only reasonable way to get a rule that can produce valid
+toolchains / platforms that need to be made available to Bazel before execution
+of any build actions.
+
+Note: this rule expects the following utilities to be installed and available on
+the PATH:
+  - docker
+  - gunzip
+  - tar
+  - sha256sum
+  - bash utilities (e.g., cp, mv, rm, etc)
+
+Known issues:
+  This rule cannot be executed inside a docker container.
+"""
+
+load("@io_bazel_rules_docker//container:pull.bzl", _pull = "pull")
+load(
+    "@bazel_toolchains//rules:version_check.bzl",
+    "check_bazel_version",
+    "extract_version_number",
+    "parse_rc",
+)
+load(
+    "@bazel_toolchains//rules:toolchain_containers.bzl",
+    "RBE_UBUNTU16_04_LATEST",
+    "public_rbe_ubuntu16_04_sha256s",
+)
+
+# External folder is set to be deprecated, lets keep it here for easy
+# refactoring
+# https://github.com/bazelbuild/bazel/issues/1262
+_EXTERNAL_FOLDER_PREFIX = "external/"
+RBE_CONFIG_DIR = "rbe_config_cc"
```

Comment:
Are these variables being exported? If not use _?

Reply:
fixed